src/pipx/standalone_python.py

```diff
@@ -0,0 +1,197 @@
+            raise PipxError(f"Unable to download python {full_version} build.") from e
+
+
+def _unpack(full_version, download_link, archive: Path, download_dir: Path):
+    with animate(f"Unpacking python {full_version} build", True):
+        # Calculate checksum
+        with open(archive, "rb") as python_zip:
+            checksum = hashlib.sha256(python_zip.read()).hexdigest()
+
+        # Validate checksum
+        checksum_link = download_link + ".sha256"
+        expected_checksum = urlopen(checksum_link).read().decode().rstrip("\n")
+        if checksum != expected_checksum:
+            raise PipxError(
+                f"Checksum mismatch for python {full_version} build. " f"Expected {expected_checksum}, got {checksum}."
+            )
+
+        with tarfile.open(archive, mode="r:gz") as tar:
+            tar.extractall(download_dir)
+
+
+def get_or_update_index():
+    """Get or update the index of available python builds from
+    the python-build-standalone repository."""
+    index_file = PIPX_STANDALONE_PYTHON_CACHEDIR / "index.json"
+    if index_file.exists():
+        index = json.loads(index_file.read_text())
+        # update index after 30 days
+        fetched = datetime.datetime.fromtimestamp(index["fetched"])
+        if datetime.datetime.now() - fetched > datetime.timedelta(days=30):
+            index = {}
+    else:
+        index = {}
+    if not index:
+        releases = get_latest_python_releases()
+        index = {"fetched": datetime.datetime.now().timestamp(), "releases": releases}
+        # update index
+        index_file.write_text(json.dumps(index))
+    return index
+
+
+def get_latest_python_releases() -> List[str]:
+    """Returns the list of python download links from the latest github release."""
+    try:
+        with urlopen(GITHUB_API_URL) as response:
+            release_data = json.load(response)
+
+    except urllib.error.URLError as e:
+        # raise
+        raise PipxError(f"Unable to fetch python-build-standalone release data (from {GITHUB_API_URL}).") from e
+
+    return [asset["browser_download_url"] for asset in release_data["assets"]]
+
+
+def list_pythons() -> Dict[str, str]:
+    """Returns available python versions for your machine and their download links."""
+    system, machine = platform.system(), platform.machine()
+    download_link_suffix = MACHINE_SUFFIX[system][machine]
+    # linux suffixes are nested under glibc or musl builds
+    if system == "Linux":
+        # fallback to musl if libc version is not found
+        libc_version = platform.libc_ver()[0] or "musl"
+        download_link_suffix = download_link_suffix[libc_version]
+
+    python_releases = get_or_update_index()["releases"]
+
+    available_python_links = [link for link in python_releases if link.endswith(download_link_suffix)]
+
+    python_versions: dict[str, str] = {}
+    for link in available_python_links:
+        match = PYTHON_VERSION_REGEX.search(link)
+        assert match is not None
```

Comment:
I don't think we should assert here, but maybe rather log a warning and skip the link?